File: PmsListenNodeService/utils/urlManager.js
/**
 * Created by 박상만 on 2017-02-22.
 */
var config = require('config-json');//('../appcfg.json');

var UrlManager = {
    pmsUrlList:{},
    gitLabUrlList:{},
    getWorkID:function (message) {
        var tmpRegex = message.match(/\[.*\]/gi);
        if(!tmpRegex){
            return "";
        }
        tmpRegex+= "";
        tmpRegex = tmpRegex.split("[").join("");
        tmpRegex = tmpRegex.split("]").join("");
        tmpRegex = tmpRegex.split(" ")[0];
        return tmpRegex;
    },

    loadConfig: function(){
        config.load('appcfg.json');
        var PMSUrl = config.get("PMSUrl");
        var PMSUrlFix = config.get("PMSUrlFix");
        var PMSAPIToken = config.get("PMSAPIToken");
        var GitUrl = config.get("GitUrl");
        var GitUrlFix = config.get("GitUrlFix");
        var GitAPIToken = config.get("GitAPIToken");
        var urlInfoPms = {Url:PMSUrl,UrlFix:PMSUrlFix,AccessToken:PMSAPIToken};
        var urlInfoGitLab = {Url:GitUrl,UrlFix:GitUrlFix,AccessToken:GitAPIToken};
        this.pmsUrlList["public"] = urlInfoPms;
        this.gitLabUrlList["public"] = urlInfoGitLab;
        console.log("Completed Load config:" + PMSUrl);
    },
    getUrlInfo:function (urlType,teamName) {
        teamName = typeof teamName !== 'undefined' ? teamName : "public";

        if(urlType == "PMSSITE" ){
            return this.pmsUrlList[teamName];
        }
        else if(urlType == "GITLABSITE" ){
            return this.gitLabUrlList[teamName];
        }
    }
};

module.exports = UrlManager;

File: PmsListenService/Util/URLManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Configuration;


namespace PmsListenService.Util
{
    public class UrlInfo
    {
        public string Url { get; set; }     //access for Server

        public string UrlFix { get; set; }  //access for user

        public string AccessToken { get; set; }  //access for user

        public UrlInfo(string url,string urlFix,string accessToken)
        {
            Url = url;
            UrlFix = urlFix;
            AccessToken = accessToken;
        }
    }

    public enum e_URLTYPE : int{
        PMSSITE = 0,
        GITLABSITE = 1
    };

    static public class URLManager
    {
        static Dictionary<string, UrlInfo> PMSUrlList = new Dictionary<string, UrlInfo>();
        static Dictionary<string, UrlInfo> GitLabUrlList = new Dictionary<string, UrlInfo>();
        static URLManager()
        {
            string PMSUrl = ConfigurationManager.AppSettings["PMSUrl"];
            string PMSUrlFix = ConfigurationManager.AppSettings["PMSUrlFix"];
            string PMSAPIToken = ConfigurationManager.AppSettings["PMSAPIToken"];

            UrlInfo pmsUrlInfo = new UrlInfo(PMSUrl, PMSUrlFix, PMSAPIToken);
            AddUrlInfo("public", e_URLTYPE.PMSSITE, pmsUrlInfo);

            string GitUrl = ConfigurationManager.AppSettings["GitUrl"];
            string GitUrlFix = ConfigurationManager.AppSettings["GitUrlFix"];

            UrlInfo gitUrlInfo = new UrlInfo(GitUrl, GitUrlFix, string.Empty);
            AddUrlInfo("public", e_URLTYPE.GITLABSITE, gitUrlInfo);

        }

        public static void AddUrlInfo(string teamName , e_URLTYPE urlType, UrlInfo urlInfo)
        {
            if(urlType == e_URLTYPE.PMSSITE)
            {
                PMSUrlList[teamName] = urlInfo;
            }
            else if (urlType == e_URLTYPE.GITLABSITE)
            {
                GitLabUrlList[teamName] = urlInfo;
            }
        }

        public static UrlInfo GetUrlInfo(e_URLTYPE urlType,string teamName = "public")
        {
            if (urlType == e_URLTYPE.PMSSITE)
            {
                return PMSUrlList[teamName];
            }
            else if (urlType == e_URLTYPE.GITLABSITE)
            {
                return GitLabUrlList[teamName];
            }
            return null;
        }

    }
}